src/main/java/org/spongepowered/api/event/entity/MoveEntityEvent.java

```diff
@@ -70,102 +72,14 @@
      * Fired when an {@link Entity}'s position changes.
      */
     @GenerateFactoryMethod
-    interface Position extends MoveEntityEvent {}
+    interface Position extends MoveEntityEvent {
+
+    }
 
     /**
-     * Fired when an {@link Entity}'s position changes for reasons other than
-     * normal movement.
+     * Fired when an {@link Entity entity's} position goes across {@link ServerWorld worlds}.
```

Comment:
I don't think it should really be named "Teleport" if this is the condition, because plugin devs associate a teleport as a non-natural, such as `/tp`, movement, and they often don't involve a world change.

A `ChangeWorldEvent` is a specialised `MoveEntityEvent` really. I would argue that if you are changing world, you should _always_ be teleporting to a specific position meaning a standalone event makes no sense. So I think that should be a sub-event, and add a `MovementType` to the event itself so plugins can inspect what type of movement caused this.

Reply:
There is the problem though.

Portal repositioning logic is VERY expensive. I fire the ChangeEntityWorld right away and allow you to change the world.

Not only that but I can't allow the Teleport to set the world as it breaks' the Mojang Teleporter so I compromised by letting you do that here (in-case you only care about the World you are going to) and then if the dev cares about the position you must cancel and do it yourself.